### PR TITLE
Update to dhtproto neo HEAD

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
The compile error
>./submodules/ocean/src/ocean/util/container/map/model/StandardHash.d(83): Error: mutable method dhtnode.node.DhtHashRange.IHashRangeListener.toHash is not callable using a const object

is caused by the `in` at https://github.com/sociomantic-tsunami/ocean/blob/v4.x.x/src/ocean/util/container/map/model/StandardHash.d#L61. This makes it impossible to use a class as a map key.

A patch release of ocean (v4.0.7) is being prepared.

**Comment by gavin-norman-sociomantic** [comment]
https://github.com/sociomantic-tsunami/ocean/pull/497

**Comment by gavin-norman-sociomantic** [comment]
Updated to use [ocean v4.0.7](https://github.com/sociomantic-tsunami/ocean/releases/tag/v4.0.7).

File: src/dhtnode/request/neo/RemoveChannel.d [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            client_name = name of client requesting channel removal
+
+        Returns:
+            true if the client is permitted to remove channels
+
+    ***************************************************************************/
+
+    override protected bool clientPermitted ( cstring client_name )
+    {
+        return client_name == "admin";
```
Comment (nemanja-boric-sociomantic) [inline src/dhtnode/request/neo/RemoveChannel.d]: Small suggestion: I suggestion making `"admin"` a class constant, instead of inlining it here.

**Pull request merged**